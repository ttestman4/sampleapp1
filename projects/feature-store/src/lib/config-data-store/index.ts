export * from './config-data-store.module';
