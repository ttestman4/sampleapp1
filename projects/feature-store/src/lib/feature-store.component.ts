import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'lib-feature-store',
  template: `
    <p>
      feature-store works!
    </p>
  `,
  styles: []
})
export class FeatureStoreComponent implements OnInit {

  constructor() { }

  ngOnInit() {
  }

}
