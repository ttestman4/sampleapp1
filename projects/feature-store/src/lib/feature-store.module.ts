import { NgModule } from '@angular/core';
import { FeatureStoreComponent } from './feature-store.component';

@NgModule({
  declarations: [FeatureStoreComponent],
  imports: [
  ],
  exports: [FeatureStoreComponent]
})
export class FeatureStoreModule { }
