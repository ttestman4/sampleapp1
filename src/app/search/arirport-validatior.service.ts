import { Injectable } from '@angular/core';
import { AbstractControl, AsyncValidator, ValidationErrors } from '@angular/forms';
import { select, Store } from '@ngrx/store';
import * as FeatuerStore from 'feature-store';
import * as NonFunctional from 'non-functional';
import { Observable, of } from 'rxjs';
import { catchError, filter, map, take } from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class ArirportValidatiorService implements AsyncValidator {

  constructor(private store: Store<FeatuerStore.ConfigData>) { }

  validate(
    control: AbstractControl
  ): Promise<ValidationErrors | null> | Observable<ValidationErrors | null> {
    return this.store.pipe(
      select(FeatuerStore.selectAirports),
      filter((airports) => airports.length > 0),
      map(airports => {
        const validCode = airports.find((ele: FeatuerStore.Airport) =>
          ele.displayText.toLowerCase() === control.value.toLowerCase());
        return NonFunctional.isUndefined(validCode) ? { 'invalidCode': { value: control.value } } : null;
      }),
      take(1),
      catchError(() => of(null))
    );
  }
}
