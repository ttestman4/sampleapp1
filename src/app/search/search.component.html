<p>
  search works!
</p>
<form>
  <mat-form-field>
    <input type="text"
    placeholder="From" aria-label="From" matInput
    [formControl]="fromCtrl"
      [matAutocomplete]="auto">
    <mat-autocomplete #auto="matAutocomplete">
      <mat-option *ngFor="let airport of airports$ | async" [value]="airport.name">
        <span>{{airport.name}}</span>|
        <small>{{airport.code}}</small>
      </mat-option>
    </mat-autocomplete>
  </mat-form-field>
  <br/>
  <mat-form-field>
    <input type="text" placeholder="Where To" aria-label="To" matInput [formControl]="fromCtrl" [matAutocomplete]="auto">
    <mat-autocomplete #auto="matAutocomplete">
      <mat-option *ngFor="let airport of airports$ | async" [value]="airport.name">
        <span>{{airport.name}}</span>|
        <small>{{airport.code}}</small>
      </mat-option>
    </mat-autocomplete>
  </mat-form-field>
</form>
