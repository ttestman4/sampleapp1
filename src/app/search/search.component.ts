import { Component, OnDestroy, OnInit } from '@angular/core';
import { FormControl } from '@angular/forms';
import { select, Store } from '@ngrx/store';
import * as FeatuerStore from 'feature-store';
import { Observable } from 'rxjs';
import { map, startWith, withLatestFrom } from 'rxjs/operators';
@Component({
  selector: 'spa-search',
  templateUrl: './search.component.html',
  styleUrls: ['./search.component.scss']
})
export class SearchComponent implements OnInit, OnDestroy {
  fromCtrl = new FormControl();
  toCtrl = new FormControl();
  airports$: Observable<FeatuerStore.Airport[]>;

  constructor(private store: Store<FeatuerStore.ConfigData>) {
    this.airports$ = this.fromCtrl.valueChanges
      .pipe(
        startWith(''),
        withLatestFrom(this.store.pipe(
          select(FeatuerStore.selectAirports)
        ), (typeAheadText, airports) => ({ typeAheadText, airports })),
        map(data => {
          return data.typeAheadText ?
            this._filterAirport(data.typeAheadText, data.airports) :
            data.airports.slice();
        })
      );
  }

  ngOnInit() {

  }

  ngOnDestroy() {
  }

  private _filterAirport(value: string, airports: FeatuerStore.Airport[]): FeatuerStore.Airport[] {
    const filterValue = value.toLowerCase();
    return airports.filter(airport => airport.name.toLowerCase().indexOf(filterValue) === 0);
  }
}
