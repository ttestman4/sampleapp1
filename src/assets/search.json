[
    {
        "origin": "BOM",
        "destination": "DEL",
        "date": "2019-11-17T00:00:00.000Z",
        "travelOrder": 1,
        "name": "indigo",
        "departureTime": {
            "hours": 9,
            "minutes": 20
        },
        "arrivalTime": {
            "hours": 16,
            "minutes": 19
        },
        "duration": {
            "hours": 7,
            "minutes": 10
        },
        "price": 1,
        "flightNo": "6E-123"
    }
]
