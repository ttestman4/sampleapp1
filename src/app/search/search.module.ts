import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { SearchComponent } from './search.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import * as Material from '@angular/material';
@NgModule({
  declarations: [
    SearchComponent
  ],
  imports: [
    CommonModule,
    FormsModule,
    ReactiveFormsModule,
    Material.MatAutocompleteModule,
    Material.MatFormFieldModule,
    Material.MatOptionModule,
    Material.MatInputModule,
    Material.MatDatepickerModule,
    Material.MatNativeDateModule,
    Material.MatSelectModule,
  ],
  exports: [
    SearchComponent
  ]
})
export class SearchModule { }
