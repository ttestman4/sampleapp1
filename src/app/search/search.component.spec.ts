// import { CommonModule } from '@angular/common';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import * as Material from '@angular/material';
import { SearchComponent } from './search.component';
import { ConfigDataStoreModule, FlightStoreModule } from 'feature-store';
import { CustomLoggerLevel, NonFunctionalModule } from 'non-functional';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ConfigureFn, configureTests } from '../../lib/testing';

describe('SearchComponent', () => {
  let component: SearchComponent;
  let fixture: ComponentFixture<SearchComponent>;

  beforeEach(
    async(() => {
      NonFunctionalModule.forTestReset();
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [SearchComponent],
          imports: [
            NoopAnimationsModule,
            FormsModule,
            ReactiveFormsModule,
            Material.MatAutocompleteModule,
            Material.MatFormFieldModule,
            Material.MatOptionModule,
            Material.MatInputModule,
            NonFunctionalModule.forRoot({
              serverLoggingUrl: '/api/logs',
              level: CustomLoggerLevel.DEBUG,
              serverLogLevel: CustomLoggerLevel.ERROR,
              disableConsoleLogging: true,
              debugStore: false
            }),
            ConfigDataStoreModule.forChild(),
            // FlightStoreModule.forRoot(),
            RouterTestingModule.withRoutes([]),
          ],
          schemas: [NO_ERRORS_SCHEMA],
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(SearchComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  xit('should create', () => {
    expect(component).toBeTruthy();
  });

  xit('snaps', () => {
    expect(fixture).toMatchSnapshot();
  });
});
