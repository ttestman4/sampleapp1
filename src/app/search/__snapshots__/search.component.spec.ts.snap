// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent snaps 1`] = `
<spa-search
  airports$={[Function Observable]}
  fromCtrl={[Function FormControl]}
  store={[Function Store]}
  toCtrl={[Function FormControl]}
>
  <p>
     search works!
  
  </p><form
    class="ng-untouched ng-pristine ng-valid"
    novalidate=""
  >
    <mat-form-field
      class="mat-form-field ng-tns-c3-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              aria-label="From"
              aria-required="false"
              autocomplete="off"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
              id="mat-input-2"
              matinput=""
              ng-reflect-autocomplete="[object Object]"
              ng-reflect-form="[object Object]"
              ng-reflect-placeholder="From"
              ng-reflect-type="text"
              placeholder="From"
              role="combobox"
              type="text"
            />
            <mat-autocomplete
              class="mat-autocomplete"
            >
              
            </mat-autocomplete>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-2"
                class="mat-form-field-label ng-tns-c3-0 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-2"
                id="mat-form-field-label-5"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                From
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c3-0 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="hint"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-tns-c3-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <br />
    <mat-form-field
      class="mat-form-field ng-tns-c3-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              aria-label="To"
              aria-required="false"
              autocomplete="off"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
              id="mat-input-3"
              matinput=""
              ng-reflect-autocomplete="[object Object]"
              ng-reflect-form="[object Object]"
              ng-reflect-placeholder="Where To"
              ng-reflect-type="text"
              placeholder="Where To"
              role="combobox"
              type="text"
            />
            <mat-autocomplete
              class="mat-autocomplete"
            >
              
            </mat-autocomplete>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-3"
                class="mat-form-field-label ng-tns-c3-1 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-3"
                id="mat-form-field-label-7"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                Where To
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c3-1 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="hint"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-tns-c3-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
  </form>
</spa-search>
`;
